wagtail/models/__init__.py

```diff
@@ -3647,24 +3647,37 @@ def start(self, workflow_state, user=None):
 
         return super().start(workflow_state, user=user)
 
+    def _user_in_groups(self, user):
+        # Cache the check whether "this user is in any of this
+        # GroupApprovalTask's groups" on the user object, in case we do it
+        # against the same user and task multiple times in a request.
+        # Use a dict to map the task id to the check result, in case we also
+        # check against different GroupApprovalTasks for the same user.
+        attr = "_group_approval_task_checks"
+        if not hasattr(user, attr):
+            setattr(user, attr, {})
+
+        if self.pk not in user._group_approval_task_checks:
+            user._group_approval_task_checks[self.pk] = self.groups.filter(
+                pk__in=user.groups.values_list("id", flat=True)
```

Comment:
Thought: As this is only one join on a _relatively small_ table, perhaps doing a `JOIN` would be more efficient here?

```suggestion
                pk__in=user.groups
```

The query behind `user.groups` ought to be simple enough.

Reply:
Not sure if I did something wrong or if Django did some magic with the query, but they seem to produce the same SQL according to debug toolbar.

I agree it's better to just use `user.groups.all()` though, as it's more readable and what the original query used to be.

```sql
SELECT '1' AS "a" FROM "auth_group" INNER JOIN "wagtailcore_groupapprovaltask_groups" ON ("auth_group"."id" = "wagtailcore_groupapprovaltask_groups"."group_id") WHERE ("wagtailcore_groupapprovaltask_groups"."groupapprovaltask_id" = '1' AND "auth_group"."id" IN (SELECT U0."id" FROM "auth_group" U0 INNER JOIN "auth_user_groups" U1 ON (U0."id" = U1."group_id") WHERE U1."user_id" = '9')) LIMIT 1
```

```sql
SELECT '1' AS "a" FROM "auth_group" INNER JOIN "wagtailcore_groupapprovaltask_groups" ON ("auth_group"."id" = "wagtailcore_groupapprovaltask_groups"."group_id") WHERE ("wagtailcore_groupapprovaltask_groups"."groupapprovaltask_id" = '1' AND "auth_group"."id" IN (SELECT U0."id" FROM "auth_group" U0 INNER JOIN "auth_user_groups" U1 ON (U0."id" = U1."group_id") WHERE U1."user_id" = '9')) LIMIT 1
```

Reply:
aha, I see the issue. It's a bit of both. The return from `values_list` is still a queryset-like object, which Django inlines into the query rather than performing 2 queries. 

If we wanted 2 queries, you'd need to wrap it in a `list` to forcefully execute the query. I agree though, `users.groups.all` is clearer.